internal/backend.go

```diff
@@ -72,6 +76,36 @@ func Run(verbose bool) {
 	select {}
 }
 
+func setupExitHandler() {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		sig := <-sigs
+		log.Printf("Received signal %d, trying to restore fan settings...", sig)
+
+		for _, controller := range Controllers {
+			for _, fan := range controller.Fans {
+				if fan.Config != nil {
+					// try to reset the pwm_enabled value
+					if fan.OriginalPwmEnabled != 1 {
+						err := setPwmEnabled(fan, fan.OriginalPwmEnabled)
+						if err == nil {
+							continue
+						}
+					}
+					err := setPwm(fan, MaxPwmValue)
+					if err != nil {
+						log.Printf("WARNING: Unable to revert fan %s, make sure it is running!", fan.Config.Id)
+					}
+				}
+			}
+		}
```

Comment:
I'd rather put this into a separate function and the rest of this setupExitHandler straight into `func main()`.
Usually we try to not hide goroutines in function calls either. Make the caller create the goroutine, in this case the `func main()`.